js/peels/sphere/to-cg.ts

```diff
@@ -66,14 +68,21 @@ function barycenterVerticesAndFaces(sphere: any, options: IOptions) {
 
     const color = options.colorFn.call(field);
 
-    positions[f * 3 + 0] = cos(fφ) * cos(fλ); // x
-    positions[f * 3 + 2] = cos(fφ) * sin(fλ); // z
-    positions[f * 3 + 1] = sin(fφ); // y
+    const x = cos(fφ) * cos(fλ);
+    const z = cos(fφ) * sin(fλ);
+    const y = sin(fφ);
+
+    positions[f * 3 + 0] = x;
+    positions[f * 3 + 1] = y;
+    positions[f * 3 + 2] = z;
 
     colors[f * 3 + 0] = color.r;
     colors[f * 3 + 1] = color.g;
     colors[f * 3 + 2] = color.b;
     colors[f * 4 + 3] = color.a != null ? color.a : 1.0;
+
+    uvs[f * 2 + 0] = 0.5 + Math.atan2(z, x) / (2 * Math.PI);
```

Comment:
Just making a note that I don't completely understand this UV mapping, but I don't know if I need to understand it. 

To me it looks like we are creating a UV map with one entry for for each field(which are X,Y,Z locations mapped over a sphere.) and setting the value of the UV map to a value computed from the X,Y,Z components of the field.

Reply:
> To me it looks like we are creating a UV map with one entry for for each field(which are X,Y,Z locations mapped over a sphere.) and setting the value of the UV map to a value computed from the X,Y,Z components of the field.

Yes, that's true.

In general in the OpenGL world, UV coordinates map each 3D vertex of a mesh to some 2D (X, Y) coordinates within the range [0, 1]. These 2D coordinates are used to tell shaders which point of a 2D texture should be used for a given mesh vertex.

When you map a rectangular plane in 3D, that's obviously pretty easy. One corner would be (0,0), another (1,0), another (1,1) and (0,1), and that's all. When we deal with a sphere, these calculations get more complex, as it's pretty close to a map projection in fact.

Also, these UV coordinates are **not** used by the colormap texture that I've added in this PR. They're used by a bump map texture (and could be used by other, standard textures that we could add to THREE.JS material).